sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -2088,6 +2098,9 @@ private void SetNameFieldsIfNull()
                     // This also makes sure that we fail fast if file doesn't exist.
                     var blobProperties = await GetPropertiesInternal(conditions: conditions, async, cancellationToken).ConfigureAwait(false);
 
+                    var expectedETag = blobProperties.Value.ETag;
+                    var readConditions = allowModifications ? new BlobRequestConditions() : new BlobRequestConditions { IfMatch = expectedETag };
```

Comment:
Why aren't these `readConditions` used on line 2113 below for the `DownloadStreamingInternal` call?

Reply:
We're passing in line 2130. The one in line 2113 belongs to delegate lambda from line 2106.

![image](https://user-images.githubusercontent.com/61715331/119524991-fb79a580-bd32-11eb-939f-ddfd85f1e7b5.png)